### PR TITLE
Bug 1303412, Quota on Registries feature

### PR DESCRIPTION
**Pull request by ahardin-rh**
This is a follow-up to https://github.com/openshift/openshift-docs/pull/1599

 It addresses this bug:
https://bugzilla.redhat.com/show_bug.cgi?id=1303412

### CODE REVIEW

**Comment by ahardin-rh** [comment]
@bfallonf @tpoitras can you please check behind me? Thanks!

File: dev_guide/builds.adoc [hunk 1]
```diff
@@ -1739,12 +1739,14 @@ manually)
...
+[cols="1,4",options="header"]
+.Troubleshooting Guidance
 |===
-
-|A build fails with `requested access to the resource is denied`. What does it mean?
+|Issue |Resolution
+|A build fails with `requested access to the resource is denied`. What does it
+mean?
 |You have exceeded one of
 link:../dev_guide/quota.html#usage-limits[image quotas] set on your project.
```
Comment (tpoitras) [inline dev_guide/builds.adoc]: s/one of/one of the

File: install_config/install/docker_registry.adoc [hunk 2]
```diff
@@ -171,17 +171,28 @@ endif::[]
...
-you may set size constraints on images, image streams and projects.
+==== Setting Storage Quota
+To protect the storage from being accidentally filled up or to prevent users
+from abusing the space, you may set size constraints on images, image streams,
+and projects.
 
-link:../../dev_guide/quota.html#create-a-quota[Create a quota object] on particular namespace with
-one or more of the following resources:
+link:../../dev_guide/quota.html#create-a-quota[Create a quota object] on
+particular namespace with one or more of the following resources:
```
Comment (bfallonf) [inline install_config/install/docker_registry.adoc]: "...on *a* particular namespace..."?

File: install_config/install/docker_registry.adoc [hunk 3]
```diff
@@ -171,17 +171,28 @@ endif::[]
...
+|Resource|Purpose
+|`openshift.io/imagesize`
+|To constrain the maximum size of an image
+
+|`openshift.io/imagestreamsize`
+|To limit the total images' size tagged in one image stream (also known as a
+repository)
+
+|`openshift.io/projectimagessize`
+|To limit the total images size per project
```
Comment (bfallonf) [inline install_config/install/docker_registry.adoc]: @ahardin-rh I'm not too sure if I understand the above two statements. Could this be simpled down to:

"To limit an image's total size when tagged in an image stream (also known as a repository)"

"To limit an image's total size per projects"

Maybe?

**Comment by bfallonf** [comment]
@ahardin-rh Just a few suggestions/questions. Looks good otherwise!

File: install_config/install/docker_registry.adoc (hunk 3 above)
Comment (tpoitras) [inline install_config/install/docker_registry.adoc]: I thought the projectimagessize meant more like "To limit the total combined size of all images per project", but I could be wrong.

File: install_config/install/docker_registry.adoc [hunk 4]
```diff
@@ -189,9 +200,8 @@ Once the quota is exceeded, an image push will fail with:
 Failed to push image: unknown: requested access to the resource is denied
 ----
 
-You may then
-link:../../admin_guide/pruning_resources.html#pruning-images[prune obsoleted images]
-to retrieve some space back.
+Then, you may link:../../admin_guide/pruning_resources.html#pruning-images[prune
```
Comment (tpoitras) [inline install_config/install/docker_registry.adoc]: s/may/can

File: install_config/install/docker_registry.adoc [hunk 5]
```diff
@@ -189,9 +200,8 @@ Once the quota is exceeded, an image push will fail with:
...
+obsoleted images] to retrieve some space back.
```
Comment (tpoitras) [inline install_config/install/docker_registry.adoc]: Unless I'm seeing this wrong in this GitHub UI, I would not put a line break between "prune" and "obsoleted images". This comes back to the discussion we had in this week's meeting about hard wrapping, where you don't want to hard wrap in the middle of a term, or link.

**Comment by tpoitras** [comment]
A few nitpicks, otherwise good @ahardin-rh :+1: :octopus: :palm_tree:

**Comment by ahardin-rh** [comment]
**revhistory note**

for link:../dev_guide/builds.html[Dev Guide --> Builds]

    Added a Troubleshooting Guidance table.

for link:../dev_guide/quota.html#usage-limits[Dev Guide --> Resource Quota]

    Added `openshift.io/imagesize`, `openshift.io/imagestreamsize`, and `openshift.io/projectimagessize` to the link:../dev_guide/quota.html#usage-limits[Usage Limits] table.

for link:../install_config/install/docker_registry.html[Deploying a Docker Registry]

    Added a link:../install_config/install/docker_registry.html#viewing-logs[Setting Storage Quota] section.

**Pull request merged**